src/main/scala/com/codacy/analysis/cli/files/FileSystemFileCollector.scala

```diff
@@ -133,6 +139,16 @@ class FileSystemFileCollector extends FileCollector[Try] {
       .map(Paths.get(_))
   }
 
+  private def checkPermissions(directory: File, files: Set[Path]): CheckedFiles = {
+    files.map(path => (path, directory / path.toString)).foldLeft(CheckedFiles(Set.empty[Path], Set.empty[Path])) {
+      case (checkedFiles, (path, file)) if file.isRegularFile && file.testPermission(PosixFilePermission.OTHERS_READ) =>
+        checkedFiles.copy(readableFiles = checkedFiles.readableFiles + path)
+      case (checkedFiles, (path, file)) =>
+        logger.error(s"Could not read file $file, make sure it is readable by everybody.")
```

Comment:
This is not an error. It is expected to happen.

Reply:
It is not expected to happen. The next phase is to fail the analysis due to this.
I was thinking a flag that can for can continuing after the one that forces permissions.